### PR TITLE
Added eclipse test case and designed more clean approach for base class

### PR DESCRIPTION
**Pull request by nitishSr**
Created application class to hold common functionalities for eclipse and codeready studio
Created simple run analysis test case for eclipse and code ready studio
Rename test case files to be separate as per IDE's
Renamed and placed common eclipse and code ready image to one folder
Some other optimizations

### CODE REVIEW

**Comment by nitishSr** [comment]
> Created application class to hold common functionalities for eclipse and codeready studio
> Created simple run analysis test case for eclipse and code ready studio
> Rename test case files to be separate as per IDE's
> Renamed and placed common eclipse and code ready image to one folder
> Some other optimizations

test_pass_result_windup_ide

File: src/tests/test_analysis_eclipse.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+
+
+def test_run_analysis(setup_eclipse):
+    """
+    Test to run a simple analysis on existing project in Eclipse IDE
+    """
+    eclipse = setup_eclipse
+    eclipse.open_mta_perspective()
+    eclipse.run_simple_analysis(project='acme')
+    assert eclipse.is_analysis_complete()
```
Comment (sshveta) [inline src/tests/test_analysis_eclipse.py:8]: Is there a way to assert on story points ?

**Review by sshveta — CHANGES_REQUESTED** [review]
Looking pretty good @nitishSr , While we are at it can we apply pre-commit checks also for the repo . Please check with Nikhil on how to do that . Since this repo is also based on python , pre commit checks will check syntax, imports etc ..

File: src/tests/test_analysis_eclipse.py (hunk 1 above)
Comment (nitishSr) [reply src/tests/test_analysis_eclipse.py:8]: You mean to say that, check story points on the report page? If so, need to discuss that.

**Comment by nitishSr** [comment]
> Looking pretty good @nitishSr , While we are at it can we apply pre-commit checks also for the repo . Please check with Nikhil on how to do that . Since this repo is also based on python , pre commit checks will check syntax, imports etc ..

@sshveta added pre commit yaml and hook and verified the same. Reference - https://github.com/pre-commit/pre-commit-hooks
pre-commit-snap

File: src/tests/test_analysis_eclipse.py (hunk 1 above)
Comment (sshveta) [reply src/tests/test_analysis_eclipse.py:8]: Yes , we can do that in later PR .

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+-   repo: https://github.com/asottile/reorder_python_imports
+    rev: v2.4.0
+    hooks:
+    -   id: reorder-python-imports
+        language_version: python3
+-   repo: https://github.com/asottile/pyupgrade
+    rev: v2.11.0
+    hooks:
+    -   id: pyupgrade
+        language_version: python3
```
Comment (sshveta) [inline .pre-commit-config.yaml:34]: https://github.com/windup/windup_integration_test/blob/master/.pre-commit-config.yaml#L7 

Add check for black too

**Review by sshveta — COMMENTED** [review]
https://github.com/windup/windup_integration_test/blob/master/.pre-commit-config.yaml#L7

Add check for black too

File: .pre-commit-config.yaml (hunk 2 above)
Comment (sshveta) [reply .pre-commit-config.yaml:34]: And in the checks section I don't see the pre-commit running . 
For the pre-commit checks to run in github you need .github/workflows/main.yaml .
Ex: https://github.com/windup/windup_integration_test/blob/master/.github/workflows/main.yml

Comment (nitishSr) [reply .pre-commit-config.yaml:34]: Made the changes.
actions-github

**Review by sshveta — APPROVED** [review]

**Pull request merged**